src/vehicle.cpp

```diff
@@ -1225,6 +1263,23 @@ bool vehicle::can_unmount( const int p, std::string &reason ) const
         return false;
     }
 
+    //Can't remove a wheel mount if there's still a wheel there
+    if( part_flag( p, "WHEEL_MOUNT_LIGHT" ) &&
```

Comment:
Can you make it a pair of two strings and iterate through collection of such pairs?

Reply:
I am not sure what you mean by this.

Reply:
Something like this, but with wheel data:
https://github.com/CleverRaven/Cataclysm-DDA/blob/60e0ca933a12015a36349f4c426472c1c39da6c6/src/weather.cpp#L924-L940

Reply:
I did that for checking wether the part can be removed.

Something similar could be done for the check wether the part can be installed in `vehicle::can_mount`. But that one is a bit messier.

I also noticed that some parts have different install and remove checks. For example alternator installing checks for `"E_ALTERNATOR"` flag but the removing check checks for `"ENGINE"` flag (probably impossible to cause problems but still a bit bad). There are also few things that require a part when installing but do not check for it when uninstalling.

The list of flags that require other flags should probably be defined in somewhere else too. Especially if the same thing is used by install and uninstall check. Maybe in JSON. But I'd rather leave that to someone else.